frontend/server/src/DAO/QualityNominations.php

```diff
@@ -190,6 +190,107 @@ private static function getVotesForNomination(int $qualitynomination_id) {
         return $votes;
     }
 
+    /**
+     * Returns the quality seal and id of a nomination for a given problem and user.
+     *
+     * @return array{quality_seal: bool, qualitynomination_id: int|null}
+     */
+    public static function getReviewedData(
+        int $problemId,
+        int $userId
+    ): array {
+        $sql = "
+            SELECT
+                qn.contents,
+                qn.qualitynomination_id
+            FROM
+                QualityNominations qn
+            INNER JOIN
+                Identities i ON i.user_id = qn.user_id
+            WHERE
+                nomination = 'quality_tag' AND
+                i.user_id = ? AND
+                qn.problem_id = ?;";
+        $params = [
+            $userId,
+            $problemId,
+        ];
+
+        /** @var array{contents: string, qualitynomination_id: int}|null */
+        $row = \OmegaUp\MySQLConnection::getInstance()->GetRow($sql, $params);
+
+        if (is_null($row)) {
+            return [
+                'quality_seal' => false,
+                'qualitynomination_id' => null,
+            ];
+        }
+
+        $contents = self::getContents($row['contents']);
+        return [
+            'quality_seal' => boolval($contents['quality_seal'] ?? false),
+            'qualitynomination_id' => $row['qualitynomination_id'],
+        ];
+    }
+
+    /**
+     * Update a QualityNominations given its id and contents
+     */
+    public static function updateQualityNominations(
+        int $qualityNominationsId,
+        string $contents
+    ): void {
+        $sqlContents = '
+            SELECT
+                contents
+            FROM
+                QualityNominations
+            WHERE
+                qualitynomination_id = ?
+            FOR UPDATE;';
+
+        $newContents = self::getContents($contents);
+
+        /** @var null|string */
+        $rowContents = \OmegaUp\MySQLConnection::getInstance()->GetOne(
+            $sqlContents,
+            [$qualityNominationsId]
+        );
+        if (!is_array($newContents) || empty($rowContents)) {
+            return;
+        }
+
+        $contentsAll = self::getContents($rowContents);
+        if (!is_array($contentsAll)) {
+            return;
+        }
+
+        $result = json_encode(array_merge($contentsAll, $newContents));
+
+        $sql = '
+            UPDATE
+                QualityNominations
+            SET
+                contents = ?
+            WHERE
+                qualitynomination_id = ?;';
+
+        \OmegaUp\MySQLConnection::getInstance()->Execute(
+            $sql,
+            [
+                $result,
+                $qualityNominationsId
+            ]
+        );
+    }
+
+    public static function getContents(string $contents): ?array {
```

Comment:
aquí lo mejor es declarar este tipo arriba del archivo:

```php
/**
 * QualityNominations Data Access Object (DAO). 
 * ...
 *
 * @psalm-type QualityNominationContents=array{tags?: mixed, before_ac?: mixed, difficulty?: mixed, quality?: mixed, quality_seal?: bool, statements?: mixed, source?: mixed, reason?: mixed, original?: mixed}
 */
```

```suggestion
    /**
     * @return null|QualityNominationContents
     */
    public static function getContents(string $contents): ?array {
```